Ollivanders/src/net/pottercraft/ollivanders2/spell/APPARATE.java

```diff
@@ -295,6 +302,31 @@ private boolean canApparateTo(@NotNull Location destination)
             return false;
         }
 
+        // check if x or z coordinates exceed max set in configuration
+        if ((destination.getX() >= p.getConfig().getInt("maxApparateDistance")) || (destination.getZ() >= p.getConfig().getInt("maxApparateDistance")))
+        {
+            common.printDebugMessage("Player is unable to teleport that far", null, null, false);
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Check to see if destination exceeds the maximum distance set in configuration
+     *
+     * @param destination the target location to apparate to
+     * @return true if the player can apparate, false otherwise
+     */
+    private boolean exceedsMaxDistance(@NotNull Location destination)
+    {
+        // check if x or z coordinates exceed max set in configuration
+        if ((destination.getX() >= p.getConfig().getInt("maxApparateDistance")) || (destination.getZ() >= p.getConfig().getInt("maxApparateDistance")))
```

Comment:
Same comment - so just in this one new feature you have it reading the config file 4 times.